functions/Invoke-DbaDbMirroring.ps1

```diff
@@ -215,8 +216,10 @@ function Invoke-DbaDbMirroring {
                     Stop-Function -Continue -Message "This mirroring configuration is not supported. Because the principal server instance, $source, is $($source.EngineEdition) Edition, the mirror server instance must also be $($source.EngineEdition) Edition."
                 }
 
-                if ($validation.MirroringStatus -ne "None") {
-                    Stop-Function -Continue -Message "Cannot setup mirroring on database ($dbName) due to its current mirroring state: $($primarydb.MirroringStatus)"
+                foreach ($status in $validation.MirroringStatus) {
```

Comment:
Why have you changed from `if` to `foreach`? You have one database, so you get one result from Invoke-DbMirrorValidation.
This has a bad effect on the -Continue in Stop-Function as this only continues the new foreach and not the old (outer) foreach.

Reply:
No, I just moved it down. I was not accounting for multiple results in an earlier test. So it would work for db1 but not db1, db2